src/mpol/datasets.py

```diff
@@ -361,6 +361,7 @@ class KFoldCrossValidatorGridded:
         q_edges (1D numpy array): an array of radial bin edges to set the dartboard cells in :math:`[\mathrm{k}\lambda]`. If ``None``, defaults to 12 log-linearly radial bins stretching from 0 to the :math:`q_\mathrm{max}` represented by ``coords``.
         phi_edges (1D numpy array): an array of azimuthal bin edges to set the dartboard cells in [radians]. If ``None``, defaults to 8 equal-spaced azimuthal bins stretched from :math:`0` to :math:`\pi`.
         npseed (int): (optional) numpy random seed to use for the permutation, for reproducibility
+        device (torch.device) : the desired device of the output. If ``None``, defalts to current device.
```

Comment:
I think the PyTorch datasets model is slightly different than nn.module, but a similar concept applies as the comments to the other modules. `UVDataset` is inherited from this class, so should be able to do this. `GriddedDataset` is not, however, so that might require a larger redesign.

Reply:
#126 relevant to any Dataset refactoring, including KFoldCrossValidator inheriting from Dataset

Reply:
removed device arg